src/transport/emitter.ts

```diff
@@ -36,19 +89,21 @@ interface EmitterFunction {
  *
  * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md
  * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md#13-content-modes
+ * @deprecated Will be removed in 4.0.0. Consider using the emitterFactory
+ *
  */
 export class Emitter {
   url?: string;
   protocol: Protocol;
-  emitter: EmitterFunction;
+  binaryEmitter: EmitterFunction;
```

Comment:
This is specifically to support the existing behavior as it is today with `Emitter`. Since a user can specify the serialization mode (BINARY or STRUCTURED), with the current `Emitter.send()`, we need to have one of each kind of emitter.

This code should all be removed with 4.0.0 given that it's currently deprecated.